mongo/integration/sdam_prose_test.go

```diff
@@ -232,4 +233,36 @@ func TestServerHeartbeatStartedEvent(t *testing.T) {
 		}
 		assert.Equal(t, expectedEvents, actualEvents)
 	})
+
+	mt := mtest.New(t)
+
+	mt.Run("polling must await frequency", func(mt *mtest.T) {
+		var heartbeatStartedCount atomic.Int64
+		servers := map[string]bool{}
```

Comment:
I'm fairly certain there are no concurrency concerns here since this is a set (i.e. nothing is aggregated).

Reply:
Go maps are not concurrent safe except for read-only use, independent of the keys or data written. Here's the data race detector error:
```
WARNING: DATA RACE
Read at 0x00c000511470 by goroutine 122:
  go.mongodb.org/mongo-driver/mongo/integration.TestServerHeartbeatStartedEvent.func2()
      /mongo-go-driver/mongo/integration/sdam_prose_test.go:266 +0x370
  go.mongodb.org/mongo-driver/mongo/integration/mtest.(*T).Run.(*T).RunOpts.func1()
      /mongo-go-driver/mongo/integration/mtest/mongotest.go:267 +0x2ac
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1689 +0x180
  testing.(*T).Run.gowrap1()
      /usr/local/go/src/testing/testing.go:1742 +0x40

Previous write at 0x00c000511470 by goroutine 195:
  runtime.mapaccess2_faststr()
      /usr/local/go/src/runtime/map_faststr.go:108 +0x42c
  go.mongodb.org/mongo-driver/mongo/integration.TestServerHeartbeatStartedEvent.func2.2()
      /mongo-go-driver/mongo/integration/sdam_prose_test.go:249 +0xb8
  go.mongodb.org/mongo-driver/x/mongo/driver/topology.(*Topology).publishTopologyDescriptionChangedEvent()
      /mongo-go-driver/x/mongo/driver/topology/topology.go:1067 +0x224
  go.mongodb.org/mongo-driver/x/mongo/driver/topology.(*Topology).apply()
      /mongo-go-driver/x/mongo/driver/topology/topology.go:973 +0xe98
  go.mongodb.org/mongo-driver/x/mongo/driver/topology.New.func1()
      /mongo-go-driver/x/mongo/driver/topology/topology.go:163 +0x78
  go.mongodb.org/mongo-driver/x/mongo/driver/topology.(*Server).updateDescription()
      /mongo-go-driver/x/mongo/driver/topology/server.go:707 +0x140
  go.mongodb.org/mongo-driver/x/mongo/driver/topology.(*Server).update.func3()
      /mongo-go-driver/x/mongo/driver/topology/server.go:624 +0xd8
  go.mongodb.org/mongo-driver/x/mongo/driver/topology.(*Server).update()
      /mongo-go-driver/x/mongo/driver/topology/server.go:654 +0x4f0
  go.mongodb.org/mongo-driver/x/mongo/driver/topology.(*Server).Connect.gowrap1()
      /mongo-go-driver/x/mongo/driver/topology/server.go:252 +0x34
```

Reply:
Yeah I see that, though for the sake of this test it seems like this shouldn't matter. Sort of similar to UDP, we're just checking that the number of heartbeats doesn't get ahead of discovery. Anyway, updated with a mutext.

Reply:
That is, even though there is a race concern I don't think you can actually break what is being tested.